slider/beatmap.py

```diff
@@ -1950,7 +1985,29 @@ def hit_objects(self,
         if sliders:
             keep_classes.append(Slider)
 
-        return tuple(ob for ob in hit_objects if
+        res = tuple(ob for ob in hit_objects if
+                    isinstance(ob, tuple(keep_classes)))
+
+        if not legacy_slider_end:
+            return res
+
+        # legacy slider end positions have the following properties:
+        # 1. the last tick of a slider is offset by -36ms from the actual end.
+        # 2. this changed offset also affects the actual end of the slider. This is already handled by the slider object.
+
+        result = deepcopy(hit_objects)
+        for i, ob in enumerate(result):
+            if isinstance(ob, Slider):
+                # 1. offset the last tick (if the slider is long enough)
+                if ob.end_time - ob.time > timedelta(milliseconds=36):
+                    temp = deepcopy(ob)
+                    temp.lazy_slider_end = True
+                    temp.tick_points = temp.tick_points
```

Comment:
I am not 100% sure, but this assignment was needed for the ticks to be re-calculated. If there is a better way that would be good, performance of this on sliderart for example is bad